internal/parser/parser.go

```diff
@@ -787,11 +787,9 @@ func (p *parser) parseTimeLiteral() *ast.DateTimeLiteral {
 func (p *parser) parseDurationLiteral() *ast.DurationLiteral {
 	pos, lit := p.expect(token.DURATION)
 	// todo(jsternberg): handle errors.
-	values, _ := ast.ParseDuration(lit)
-	return &ast.DurationLiteral{
-		Values:   values,
-		BaseNode: p.posRange(pos, len(lit)),
-	}
+	node, _ := ParseDuration(lit)
```

Comment:
I would prefer at the minimum that the duration parser in the internal api not change. It should return `[]ast.Duration` as that's what it parses. I know this implementation works, but I don't see a reason why this needs  to change to be different from the others and I prefer the old pattern since it makes type safety more useful by not partially creating a value.

Reply:
As I understand it the concern is that returning ast.DurationLiteral leaves the position information incomplete. That makes sense. What about the `Source` field of the ast.SourceLocation, I would expect this method to populate that, or should we break that into two separate methods? One that parser the values and the other that creates the ast.Node with the SourceLocation.Source populated?

Reply:
The `Source` field is part of the position information and gets filled in the same way. That's how it works currently.

Reply:
But if these functions do not return an ast.Node then the caller has to know how to fill in the source information. I think a godoc stating that only the source is filled in while the rest of the position info is left blank is fine. It meets the current needs and is clear how it behaves.

Reply:
Ah, I didn't understand what you meant. The source is filled in directly from a byte slice anyway  so the two are tied to each other. In the current implementation, the source attribute is not filled in by the input to this method.

I think it gets confusing to separate these two and that's not how the current implementation does it so I would prefer that remain unchanged.

Reply:
Yeah, I think the current implementation in this PR is in an odd middle ground. It should either return the raw values or return ast.Node with the Source data filled in.

I am fine with either.